### ISSUE TITLE
Support for Embeddings

### ISSUE THREAD

**Issue by slashmili** [brainlid/langchain#40]
Hey there!

I'd like to see LangChain supports for [embeddings](https://api.python.langchain.com/en/latest/api_reference.html#module-langchain.embeddings)! Probably first the [OpenAPI's](https://platform.openai.com/docs/guides/embeddings) embeddings and then [HuggingFace](https://huggingface.co/blog/getting-started-with-embeddings) and others

Is this something that you have thought about and have a plan? I'd love to get some guidance and give it a try.

> **Disclaimer**: I'm new to this field, if I say something that doesn't make sense please point out.

**Comment by brainlid** [comment]
Hello @slashmili! I don't have any immediate plans in that direction. Yes, I believe that vectorstores are needed as well as some foundational pieces. 

I'd love to have support for this for use with Agents or Assistants. However, I'm coming at it from a different direction and working on building more advanced Assistants (which is valuable alone) before focusing on the document embeddings. 

Unfortunately, I can't offer much guidance in this area either. :slightly_smiling_face:

**Comment by slashmili** [comment]
Thanks for your feedback, I close this issue for now.

I'll try to fill my knowledge gap and come back to you again in few months 😃

**Comment by cpursley** [comment]
Is there any more interest in this? It would be neat to pair up with [pgvector](https://github.com/pgvector/pgvector). Here's a neat talk on the subject: https://www.youtube.com/watch?v=RFnZB76KVWk

**Comment by brainlid** [comment]
@cpursley, I agree it would very cool and I would love to have it in. However, I don't have the bandwidth for it and I don't have a personal need for it at the moment. 

I'm open to PRs!

**Comment by slashmili** [comment]
@brainlid is there a good place to have the discussion? e.g Elixir Slack, Elixir Forum, Elixir Discord server or here in the discussion section of GitHub repo. I have some idea I'd like to share and get some feedback before implementing anything.

**Comment by cpursley** [comment]
I'd also love a place do discuss langchain. My suggestion is a channel in the Elixir slack (I really don't like discord).
